### PR TITLE
Encounter: {Get/Set}Reset

### PR DESCRIPTION
**Pull request by hendrikgit**

### CODE REVIEW

File: Plugins/Encounter/Encounter.cpp [hunk 1]
```diff
@@ -190,6 +190,29 @@ ArgumentStack Encounter::SetPlayerTriggeredOnly(ArgumentStack&& args)
     return Services::Events::Arguments();
 }
 
+ArgumentStack Encounter::GetReset(ArgumentStack&& args)
```
Comment (mtijanic) [inline Plugins/Encounter/Encounter.cpp:193]: These two also need to be added to the `REGISTER()` list at the top of the file

File: Plugins/Encounter/NWScript/nwnx_encounter.nss [hunk 2]
```diff
@@ -53,6 +53,16 @@ int NWNX_Encounter_GetPlayerTriggeredOnly(object encounter);
 /// @param playerTriggeredOnly TRUE/FALSE
 void NWNX_Encounter_SetPlayerTriggeredOnly(object encounter, int playerTriggeredOnly);
 
+/// @brief Get if the encounter respawns or not.
+/// @param encounter The encounter object.
+/// @return TRUE if the encounter does respawn, FALSE otherwise.
+int NWNX_Encounter_GetReset(object encounter);
```
Comment (mtijanic) [inline Plugins/Encounter/NWScript/nwnx_encounter.nss:59]: The function name is not very descriptive.. maybe `{G,S}etRespawning

File: Plugins/Encounter/Encounter.cpp (hunk 1 above)
Comment (hendrikgit) [reply Plugins/Encounter/Encounter.cpp:195]: Oops. Thanks. Added now.

File: Plugins/Encounter/NWScript/nwnx_encounter.nss (hunk 2 above)
Comment (hendrikgit) [reply Plugins/Encounter/NWScript/nwnx_encounter.nss:59]: I thought about that too but at the same time that just seems to be it's name.
The member variable is called reset and so is the field in the gff. There are also the two related functions GetResetTime, SetResetTime.
People who are familiar with the gff might look/search for something with reset.

Should I change it anyway?

Comment (mtijanic) [reply Plugins/Encounter/NWScript/nwnx_encounter.nss:59]: For what it's worth, the internal name is actually `{G,S}etEncounterCanReset()`, it's just the functions are not exposed in the API, only the field. I think something GetReset() is too confusing. GetCanReset() would be much better.

Comment (hendrikgit) [reply Plugins/Encounter/NWScript/nwnx_encounter.nss]: Changed it to GetCanReset, SetCanReset.

**Pull request merged**